distribution/helm/vector/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,11 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "vector.serviceAccountName" . }}
+  labels:
+    app.kubernetes.io/name: {{ template "vector.name" . }}
```

Comment:
I believe helm allows defining a helper with these labels and use them everywhere. AFAIK helm3 does this by default.